src/stactools/naip/utils.py

```diff
@@ -59,3 +68,24 @@ def _parse(lines, group_indent=0):
             return result
 
     return _parse(md_text.split("\n"), group_indent=0)["Metadata"]
+
+
+def missing_element(attribute: str) -> Callable[[str], Exception]:
+    def get_exception(xpath: str) -> Exception:
+        return MissingElement(
+            f"Could not find attribute `{attribute}` at xpath '{xpath}'."
+        )
+
+    return get_exception
+
+
+def get_id_date(cog_href) -> Tuple[str, datetime]:
```

Comment:
```suggestion
def maybe_extract_id_and_date(cog_href: str) -> Optional[Tuple[str, datetime]]:
```

I use the convention of using "maybe" with a partially-defined function (e.g., this  can't extract an id and dt from every string, only valid NAIP ids.

I avoid `get` except for pure getter methods, e.g., returning a value, and use "extract", "retreive", "process", etc. for other methods that are doing computation.